renderer/src/utils.rs

```diff
@@ -23,3 +23,53 @@ pub(crate) fn derive_normal_matrix(m: &Matrix4) -> Matrix3 {
         .unwrap_or_else(Matrix3::identity)
         .transpose()
 }
+
+
+#[cfg(test)]
+mod tests {
+    use ldraw::{Matrix4, Matrix3};
+
+    use super::{truncate_matrix4, derive_normal_matrix};
+
+    #[test]
+    fn test_truncate_matrix4() {
+        let matrix = Matrix4::new(
+            1.0, 2.0, 3.0, 4.0,
+            5.0, 6.0, 7.0, 8.0,
+            9.0, 10.0, 11.0, 12.0,
+            13.0, 14.0, 15.0, 16.0
+        );
+
+        let truncated = truncate_matrix4(&matrix);
+
+        assert_eq!(
+            truncated,
+            Matrix3::new(
+                1.0, 2.0, 3.0,
+                5.0, 6.0, 7.0,
+                9.0, 10.0, 11.0
+            )
+        )
+    }
+
+    #[test]
+    fn test_derive_normal_matrix_1() {
```

Comment:
This name implies that there should be another subsequent tests on `derive_normal_matrix` (`test_derive_normal_matrix_2`, ...) Is that intentional?

Reply:
I originally intended to do so, but it seems that a single test is sufficient 😓  I have edited the test name in dba6e30. Thanks! 👍